### ISSUE TITLE
Support negative values

### ISSUE THREAD

**Issue by jason-green-io** [tenox7/ttyplot#53]
This would be great to plot temperature, or other values that typically sometimes go negative.

**Comment by tenox7** [comment]
ack I will work on that

**Comment by hartwork** [comment]
Related to #52

**Comment by tenox7** [comment]
implemented in head/master

**Comment by tenox7** [comment]
testing would be appreciated

**Comment by tenox7** [comment]
👁️ ah I see it now thank you for pointing it out!

**Comment by tenox7** [comment]
I think I have a working fix for this in https://github.com/tenox7/ttyplot/commit/f3bfe02843c9e64ef8398a8c3f6399ba867ce6e2

I also added printing 0 values when negative is present. I was debating this for long time but it appears to be standard. It looks little odd but I think it's valid. I'd appreciate your review and opinion.

<img width="777" height="415" alt="Image" src="https://github.com/user-attachments/assets/88bec5f5-b6b0-411c-8296-4625e3af900f" />

**Comment by edgar-bonet** [comment]
@tenox7 wrote:
> I'd appreciate your review and opinion.

Nice job!

Now, negative values seem to work fine. I do also find that that the _z_ axis being “thick” looks odd. I would rather have it with zero width, but that may be just personal preference.

There may be a rounding issue going on here: the bottom of the plot seems to have been cropped. In the plot you showed, I noticed that the bottom of the sine wave is narrower than the top, and actually has the same with as the row just below the top. See the pale blue measurements:

<img width="777" height="415" alt="Image" src="https://github.com/user-attachments/assets/a50cb52e-fb43-4954-af7b-2417dc0ab326" />

I may just be nitpicking...

**Comment by tenox7** [comment]
Yeah... I debated this and did some research. In most or all plot systems zero values, or zero crossing points do get plotted. The issue is that in graphical plots they are 1 pixel height or something negligible. 

Perhaps as an enhancement a zero line could be printed on the screen with ----------- characters. However technically it's better to see if a value had 0 value or not. It's possible that a value could have 0 value or not have zero value at some sample. Its better to see it than have it smeared with -----. I think.

Also for the bottom side being cropped I did notice that. I don't like it, but I would like to release with this and fix this rounding on bottom in a latter release.

**Comment by edgar-bonet** [comment]
@tenox7 wrote:
> Also for the bottom side being cropped I did notice that. I don't like it, but I would like to release with this and fix this rounding on bottom in a latter release.

Makes a lot of sense to me.
